### PR TITLE
Forward port #2810

### PR DESCRIPTION
**Pull request by fballiano**
#2810 was left out of #2766 because of conflicts, here's the merge in its own PR, please check :-)

### CODE REVIEW

File: app/code/core/Mage/Sales/Model/Order.php [hunk 1]
```diff
@@ -441,42 +441,42 @@ class Mage_Sales_Model_Order extends Mage_Sales_Model_Abstract
     protected $_eventObject = 'order';
 
     /**
-     * @var Mage_Sales_Model_Resource_Order_Address_Collection|Mage_Sales_Model_Order_Address[]|null
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order.php:444]: keep as is in v20

File: app/code/core/Mage/Sales/Model/Order.php [hunk 2]
```diff
@@ -519,11 +519,7 @@ protected function _construct()
     protected function _initOldFieldsMap()
     {
         // pre 1.6 fields names, old => new
-        $this->_oldFieldsMap = [
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order.php:522]: keep as is in v20

File: app/code/core/Mage/Sales/Model/Order.php [hunk 3]
```diff
@@ -1745,15 +1741,15 @@ public function getAllVisibleItems()
         $items = [];
         foreach ($this->getItemsCollection() as $item) {
             if (!$item->isDeleted() && !$item->getParentItemId()) {
-                $items[] = $item;
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order.php:1748]: keep as is in v20

File: app/code/core/Mage/Sales/Model/Order.php [hunk 4]
```diff
@@ -1745,15 +1741,15 @@ public function getAllVisibleItems()
...
+                $items[] =  $item;
             }
         }
         return $items;
     }
 
     /**
      * @param int $itemId
-     * @return Mage_Sales_Model_Order_Item|null
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order.php:1756]: keep as is in v20 (and maybe backport this change to v19?)

File: app/code/core/Mage/Sales/Model/Order/Creditmemo/Item.php [hunk 5]
```diff
@@ -144,9 +144,7 @@ public function setCreditmemo(Mage_Sales_Model_Order_Creditmemo $creditmemo)
     protected function _initOldFieldsMap()
     {
         // pre 1.6 fields names, old => new
-        $this->_oldFieldsMap = [
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order/Creditmemo/Item.php:147]: keep as is in v20

File: app/code/core/Mage/Sales/Model/Order/Invoice/Item.php [hunk 6]
```diff
@@ -115,14 +111,12 @@ public function _construct()
      * Init mapping array of short fields to
      * its full names
      *
-     * @return $this
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order/Invoice/Item.php:118]: keep as is in v20, and backport to v19

File: app/code/core/Mage/Sales/Model/Order/Invoice/Item.php [hunk 7]
```diff
@@ -115,14 +111,12 @@ public function _construct()
...
+     * @return Varien_Object
      */
     protected function _initOldFieldsMap()
     {
         // pre 1.6 fields names, old => new
-        $this->_oldFieldsMap = [
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order/Invoice/Item.php:123]: keep as is in v20

File: app/code/core/Mage/Sales/Model/Order/Item.php [hunk 8]
```diff
@@ -238,13 +237,12 @@ protected function _construct()
      * its full names
      *
      * @return Varien_Object
+     * @deprecated
      */
     protected function _initOldFieldsMap()
     {
         // pre 1.6 fields names, old => new
-        $this->_oldFieldsMap = [
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order/Item.php:245]: keep as is in v20

**Comment by tmotyl** [comment]
added comments, most of the changes should not be applied on v20.

**Comment by fballiano** [comment]
@sreichel reading @tmotyl comments, IMHO we should revert #2810, it was done for an easier sync and it seems it's making it more problematic.

**Comment by tmotyl** [comment]
@fballiano I don't see a point in reverting https://github.com/OpenMage/magento-lts/pull/2810 .
This PR mixes changes which were made in v20 branch on purpose (e.g. OldFieldsMap) with ones that makes sense to be ported to v20. And some (typehints) which make sense to port from v20 to v19.

**Comment by fballiano** [comment]
@tmotyl I think I fixed all your comments. At this point I don't understand what this PR was about.

**Comment by fballiano** [comment]
i'm closing this one since i don't know wth i should do

**Pull request closed without merging**